consensus/core/src/block.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    cell::OnceCell,
+    hash::{Hash, Hasher},
+};
+
+use fastcrypto::hash::HashFunction;
+use serde::{Deserialize, Serialize};
+
+use consensus_config::{AuthorityIndex, DefaultHashFunction};
+
+/// Round number of a block.
+pub type Round = u32;
+
+/// Block proposal timestamp in milliseconds.
+pub type BlockTimestampMs = u64;
+
+/// One validator can produce at most one block per round.
+/// A block includes references to previous round blocks and transactions that the validator
+/// considers valid.
+#[derive(Clone, Deserialize, Serialize)]
+pub enum Block {
+    V1(BlockV1),
+}
+
+pub trait BlockAPI {
+    fn reference(&self) -> BlockRef;
+    fn digest(&self) -> BlockDigest;
+    fn round(&self) -> Round;
+    fn author(&self) -> AuthorityIndex;
+    fn timestamp(&self) -> BlockTimestampMs;
+    fn includes(&self) -> &[BlockRef];
+    // TODO: add accessor for transactions.
+}
+
+#[derive(Clone, Default, Deserialize, Serialize)]
+pub struct BlockV1 {
+    round: Round,
+    author: AuthorityIndex,
+    timestamp: BlockTimestampMs,
```

Comment:
Since this timestamp is used to compute the commit timestamp, I feel it is better to not include how the timestamp is generated (`creation_`). We can come back to fix this later if `timestamp` is confusing.